sources/packages/backend/apps/api/src/services/announcements/announcements.service.ts

```diff
@@ -0,0 +1,33 @@
+import { Injectable } from "@nestjs/common";
+import { Announcements, RecordDataModelService } from "@sims/sims-db";
+import { DataSource, LessThanOrEqual, MoreThanOrEqual } from "typeorm";
+
+@Injectable()
+export class AnnouncementsService extends RecordDataModelService<Announcements> {
+  constructor(dataSource: DataSource) {
+    super(dataSource.getRepository(Announcements));
+  }
+
+  /**
+   * Get system announcements.
+   * @returns system announcements list.
+   */
+  async getAnnouncements(): Promise<Announcements[]> {
+    const now = new Date();
+    return this.repo.find({
+      select: {
+        id: true,
+        messageTitle: true,
+        message: true,
+        target: true,
+        startDate: true,
+        endDate: true,
```

Comment:
As we are selecting all the column, we can ignore the select like this. Please double-check.

![image](https://github.com/user-attachments/assets/af42878f-19c7-40e9-8625-f838263c2f33)

Reply:
updated